### PR TITLE
add python packaging support for linux-x86-64

### PR DESCRIPTION
**Pull request by arroyc**
will add the UT shortly ...

### CODE REVIEW

File: src/BuildScriptGenerator/Helpers/TemplateHelper.cs [hunk 1]
```diff
@@ -109,11 +109,6 @@ public static TemplateResource NodeBuildSnippet
                 get => new TemplateResource("Microsoft.Oryx.BuildScriptGenerator.Node.NodeBashBuildSnippet.sh.tpl");
             }
 
-            public static TemplateResource NodePackageSnippet
-            {
```
Comment (arroyc) [inline src/BuildScriptGenerator/Helpers/TemplateHelper.cs:113]: dead code

File: src/BuildScriptGenerator/Node/NodeBashPackageSnippet.sh.tpl [hunk 2]
```diff
@@ -1,2 +0,0 @@
-﻿npm run prepare || true
```
Comment (arroyc) [inline src/BuildScriptGenerator/Node/NodeBashPackageSnippet.sh.tpl:1]: dead code

File: tests/Oryx.BuildImage.Tests/Python/PythonPackageTest.cs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+                osReqsParam = $"--os-requirements {string.Join(',', requiredOsPackages)}";
+            }
+            
+            // pypi package url usually is in following format
+            //https://pypi.io/packages/source/{ package_name_first_letter }/{ package_name }/{ package_name }-{ package_version }.tar.gz
+
+            var pkgNameFirstLetter = pkgName.ElementAt(0);
+            var pyPiTarUrl = $"https://pypi.io/packages/source/{pkgNameFirstLetter}/{pkgName}/{pkgName}-{pkgVersion}.tar.gz";
+
+            if (string.IsNullOrEmpty(pkgTag))
```
Comment (kichalla) [inline tests/Oryx.BuildImage.Tests/Python/PythonPackageTest.cs:75]: nit: braces

File: tests/Oryx.BuildImage.Tests/Python/PythonPackageTest.cs [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                pkgTag = pkgVersion;
+
+            var script = new ShellScriptBuilder()
+            // Fetch source code
+                .AddCommand($"mkdir -p {pkgSrcDir} && git clone {gitRepoUrl} {pkgSrcDir}")
```
Comment (kichalla) [inline tests/Oryx.BuildImage.Tests/Python/PythonPackageTest.cs:80]: Check this to make this faster: https://stackoverflow.com/a/39067940/1184056

File: src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl [hunk 5]
```diff
@@ -104,6 +104,18 @@ fi
 	echo $APP_PACKAGES_PATH > $SITE_PACKAGES_PATH"/oryx.pth"
 {{ end }}
 
+{{ if RunPythonPackageCommand }}
```
Comment (kichalla) [inline src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl:107]: Shouldn't this be mutually exclusive with the other things that we do in this tpl file? should we move this package stuff to a separate file if it improves readability?

File: tests/Oryx.BuildImage.Tests/Python/PythonPackageTest.cs [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+            // Assert contained file names
+            var tarLists = result.StdOut.Split(tarListMarker);
+
+            var (oryxTarList, oryxTarSize) = ParseTarList(tarLists[1]);
+            var (pypiTarList, pypiTarSize) = ParseTarList(tarLists[2]);
+
+            var unContained = pypiTarList.Where(x => !oryxTarList.Contains(x));
+            Assert.Equal(pypiTarList, oryxTarList);
+            
+
```
Comment (kichalla) [inline tests/Oryx.BuildImage.Tests/Python/PythonPackageTest.cs:113]: nit: extra line..similarly other places in this file

File: tests/Oryx.BuildImage.Tests/Python/PythonPackageTest.cs [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+            var pyPiTarUrl = $"https://pypi.io/packages/source/{pkgNameFirstLetter}/{pkgName}/{pkgName}-{pkgVersion}.tar.gz";
+
+            if (string.IsNullOrEmpty(pkgTag))
+                pkgTag = pkgVersion;
+
+            var script = new ShellScriptBuilder()
+            // Fetch source code
+                .AddCommand($"mkdir -p {pkgSrcDir} && git clone {gitRepoUrl} {pkgSrcDir}")
+                .AddCommand($"cd {pkgSrcDir} && git checkout tags/{pkgTag} -b test/{pkgVersion}")
+                .AddCommand("source benv python=3")
```
[the remark this answers is not in the document]
Comment (arroyc) [reply tests/Oryx.BuildImage.Tests/Python/PythonPackageTest.cs:82]: which one are you referring here? benv command?

File: src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl (hunk 5 above)
Comment (arroyc) [reply src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl:107]: I thought about it .. but then I did a quick referrence in node .. and seems like it's in the same file .. so I keep it same

**Pull request merged**